nixos/modules/services/web-servers/nginx/default.nix

```
@@ -348,16 +348,25 @@ let
         # The acme-challenge location doesn't need to be added if we are not using any automated
         # certificate provisioning and can also be omitted when we use a certificate obtained via a DNS-01 challenge
         acmeLocation = optionalString (vhost.enableACME || (vhost.useACMEHost != null && config.security.acme.certs.${vhost.useACMEHost}.dnsProvider == null)) ''
+          # See https://gist.github.com/mingderwang/bdf30ead416c9dc0b29710f864c88517 for source of this block
+          # Hide /acme-challenge subdirectory and return 404 on all requests.
+          # It is somewhat more secure than letting Nginx return 403.
+          # Ending slash is important!
```

```suggestion
          # Hide /.well-known/acme-challenge directory and return 404 on all requests
```
for nginx rules the slashes are always important.

the linked source is not that great. It links two other sources which both do not contain the part we are looking for.